src/server/pfs/server/master_v2.go

```diff
@@ -21,20 +21,24 @@ const (
 func (d *driverV2) master(env *serviceenv.ServiceEnv, objClient obj.Client, db *gorm.DB) {
 	ctx := context.Background()
 	masterLock := dlock.NewDLock(d.etcdClient, path.Join(d.prefix, masterLockPath))
-	err := backoff.RetryNotify(func() error {
-		masterCtx, err := masterLock.Lock(ctx)
-		if err != nil {
+	for {
```

Comment:
This shouldn't need a for loop wrapping it. If `Run` is returning nil, then that is a problem that needs to be fixed (I vaguely remember this being an issue that had to do with the context cancel not being propagated).

Reply:
Yeah I thought this was fixed.  `gc.Run` should never return right?

Reply:
Discussed offline. Will add a TODO here. This may be related to the environment being torn down.